src/schedule/ll_schedule.c

```diff
@@ -230,15 +225,36 @@ static int schedule_ll_domain_set(struct ll_schedule_data *sch,
 
 	spin_lock(&sch->domain->lock);
 
+	tr_dbg(&ll_tr, "task->start %u next_tick %u",
+	       (unsigned int)task->start,
+	       (unsigned int)sch->domain->next_tick);
+
+	task_start_us = period ? period : start;
+	task_start_ticks = sch->domain->ticks_per_ms * task_start_us / 1000;
+	task_start = task_start_ticks + platform_timer_get(timer_get());
+
+	if (sch->domain->next_tick == UINT64_MAX || !period) {
+		/* first or one shot task, set domain */
+		domain_set(sch->domain, task_start);
+		task->start = sch->domain->next_tick;
+	} else if (sch->domain->next_tick <= task_start) {
+		/* the task will be covered by the coming interrupt */
+		task->start = sch->domain->next_tick;
+	} else {
+		/* earlier periodic task, try to make it cadence aligned with the existed task */
+		offset = (sch->domain->next_tick - task_start) %
+			 sch->domain->ticks_per_ms * period / 1000;
```

Comment:
are you sure this is correct? Did you really mean `offset = ((sch->domain->next_tick - task_start) % sch->domain->ticks_per_ms) * period / 1000;`?

Reply:
> are you sure this is correct? Did you really mean `offset = ((sch->domain->next_tick - task_start) % sch->domain->ticks_per_ms) * period / 1000;`?

Oh good catch, thanks. I meant `offset = (sch->domain->next_tick - task_start) % (sch->domain->ticks_per_ms * period / 1000);`.